### DISCUSSION TITLE
how to calculate the tint value to use

### DISCUSSION THREAD

**Discussion by srini8080** [ImageMagick/ImageMagick#3975] [Digital Image Processing]
![8](https://user-images.githubusercontent.com/12046302/126731813-51de9d15-1227-4011-9e6e-7689ec4561f8.jpg)
![64](https://user-images.githubusercontent.com/12046302/126731827-af1811c9-b4b6-47ce-8fe9-73fa0913062a.jpg)

I have 2 images. I am trying to remove the Tan of these images which are scanned pages (sun burnt)

convert 8.jpg -fill white -tint 110 -channel rgb -auto-level -sharpen 0x10 x.jpg

If tint value 110 is given to the first one, then the image looks good.

If a tint value 130 is given to the second one, then the image looks good.

How do I automatically calculate the tint based on the image?

Thanks

**Comment by fmw42** [comment]
What platform? What version of ImageMagick? (Please always provide that information)

I would do it differently. Define a small patch of background (not too close to the corner but any place you want that is background color). Say 20x20 at +20+20 offset. Get the average color of the patch. Then divide that color into the image. You can add -channel rgb -auto-level as desired.

Unix syntax:

```
patch="20x20+20+20"
color=`convert 8.jpg[$patch] +repage -scale 1x1! -format "%[pixel:u.p{0,0}]" info:`
convert 8.jpg \( +clone -fill "$color" -colorize 100 \) +swap -compose divide -composite result.jpg
```

(Sorry, I do not know Windows syntax enough to provide the variable).

![result](https://user-images.githubusercontent.com/13458218/126734021-fbeefee2-8b58-4af7-b823-92c28c33e3de.jpg)

![result](https://user-images.githubusercontent.com/13458218/126734116-a408664b-7560-4eb2-bbfc-f58e3029c9ba.jpg)

**Reply by srini8080** [reply]
Sorry, I forget this everytime ;-(

Version: ImageMagick 6.9.10-71 Q16 x86_64 2021-07-06 https://imagemagick.org
Copyright: © 1999-2019 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC OpenMP(3.1) 
Delegates (built-in): bzlib fftw fontconfig freetype jng jp2 jpeg lzma openexr pangocairo png tiff x xml zlib

Centos 7.6

**Reply by srini8080** [reply]
Also, the reason I started using tint is that for tint value of 145, the enclosed image is cleaned.

The next question I was going to pose is, how to do that for grayscale images with noise?

![5](https://user-images.githubusercontent.com/12046302/126736975-89d48f1b-ef0c-41ba-84c3-e520ec12f507.jpg)

**Comment by fmw42** [comment]
Use local area thresholding. (Or my script, textcleaner)

`convert gray.jpg -negate -lat 10x10+10% -negate result.jpg`

![result](https://user-images.githubusercontent.com/13458218/126737272-ca115659-5575-48b4-a60b-e25aed2a4619.jpg)

**Reply by srini8080** [reply]
convert 5.jpg -fill white -tint 145 -channel rgb -auto-level x.jpg is giving me better result
![x](https://user-images.githubusercontent.com/12046302/126738052-4f5d8aa4-c517-4443-9d60-090dd9363a23.jpg)

Thanks,

**Reply by srini8080** [reply]
Also for images like this, I can adjust the tint value (If I know that it is clean and no noise)

![4](https://user-images.githubusercontent.com/12046302/126738091-b769e29d-fe59-4c9a-8135-728500b90de5.jpg)
